modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/APIResourceManagementClient.java

```diff
@@ -0,0 +1,296 @@
+
+        String endPointUrl = apiResourceManagementBasePath;
+        if (StringUtils.isNotBlank(apiResourceFilter)) {
+            endPointUrl = apiResourceManagementBasePath + "?filter=" + apiResourceFilter;
+        }
+        try (CloseableHttpResponse response = getResponseOfHttpGet(endPointUrl, getHeadersWithBasicAuth())) {
+            String responseBody = EntityUtils.toString(response.getEntity());
+            ObjectMapper jsonWriter = new ObjectMapper(new JsonFactory());
+            APIResourceListResponse apiResourceListResponse =
+                    jsonWriter.readValue(responseBody, APIResourceListResponse.class);
+            return apiResourceListResponse.getApiResources();
+        }
+    }
+
+    /**
+     * Get API resources by filtering from a sub organization.
+     *
+     * @param apiResourceFilter API resource filter.
+     * @param switchedM2MToken Switched M2M token.
+     * @return List of API resources.
+     * @throws Exception Error when getting the filtered API resources.
+     */
+    public List<APIResourceListItem> getAPIResourcesWithFilteringFromSubOrg(String apiResourceFilter,
+                                                                            String switchedM2MToken) throws Exception {
+
+        String endPointUrl = getSubOrgAPIResourceMgtPath();
+        if (StringUtils.isNotBlank(apiResourceFilter)) {
+            endPointUrl = endPointUrl + "?filter=" + apiResourceFilter;
+        }
+        try (CloseableHttpResponse response = getResponseOfHttpGet(endPointUrl,
+                getHeadersWithBearerToken(switchedM2MToken))) {
+            String responseBody = EntityUtils.toString(response.getEntity());
+            ObjectMapper jsonWriter = new ObjectMapper(new JsonFactory());
+            APIResourceListResponse apiResourceListResponse =
+                    jsonWriter.readValue(responseBody, APIResourceListResponse.class);
+            return apiResourceListResponse.getApiResources();
+        }
+    }
+
+    /**
+     * Get API resource scopes.
+     *
+     * @param apiId API Id.
+     * @return List of API resource scopes.
+     * @throws IOException Error when getting the scopes.
+     */
+    public List<ScopeGetModel> getAPIResourceScopes(String apiId) throws IOException {
+
+        String endPointUrl = apiResourceManagementBasePath + PATH_SEPARATOR + apiId;
+        try (CloseableHttpResponse response = getResponseOfHttpGet(endPointUrl, getHeadersWithBasicAuth())) {
+            String responseBody = EntityUtils.toString(response.getEntity());
+            ObjectMapper jsonWriter = new ObjectMapper(new JsonFactory());
+            APIResourceResponse apiResourceResponse = jsonWriter.readValue(responseBody, APIResourceResponse.class);
+            return apiResourceResponse.getScopes();
+        }
+    }
+
+    /**
+     * Get API resource scopes from a sub organization.
+     *
+     * @param apiId API Id.
+     * @param switchedM2MToken Switched M2M token.
```

Comment:
Let's fix formatting here

Reply:
check similar places

Reply:
fixed in [fea9dce](https://github.com/wso2/product-is/pull/23306/commits/fea9dce37c558217627f44ca67866f3ed22d81f9)